### PR TITLE
Migrating ContainerStorage.Management to latest TypeSpec

### PR DESCRIPTION
**Pull request by allenjzhang**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
              Migrating to latest TypeSpec

Changes:
1. Correctly set up folder structure
2. Updated example files
3. source formatting
4. swagger is updated with mostly reordering source code
5. There are two ListBySubscription route showed up due to ARM lib's `ResourceCollectionOperations` since has added this standard operation.

### CODE REVIEW

File: specification/containerstorage/ContainerStorage.Management/tspconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+emit:
+  - '@azure-tools/typespec-autorest'
+options:
+  '@azure-tools/typespec-autorest':
+    emitter-output-dir: "{project-root}/.."
+    azure-resource-provider-folder: "resource-manager"
+    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/containerstorage.json"
+    examples-directory: "{project-root}/examples"
```
Comment (mikeharder) [inline specification/containerstorage/ContainerStorage.Management/tspconfig.yaml:8]:
```suggestion
    examples-directory: "{project-root}/examples"

```

Comment (mikeharder) [inline specification/containerstorage/ContainerStorage.Management/tspconfig.yaml:8]:
```suggestion
    azure-resource-provider-folder: "resource-manager"
    emitter-output-dir: "{project-root}/.."
    examples-directory: "{project-root}/examples"
    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/containerstorage.json"
```

**Review by mikeharder — CHANGES_REQUESTED** [review]
This is failing CI due to some deprecation messages:

```
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:27:23 - error deprecated: Deprecated: object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.
> 27 |   TProperties extends object,
     |                       ^^^^^^
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:180:6 - error deprecated: Deprecated: Template constrainted to 'string' will not be assignable to 'valueof string' in the future. Update the constraint to be 'valueof string'
> 180 | @doc(TMessage)
      |      ^^^^^^^^
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:40:23 - error deprecated: Deprecated: object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.
> 40 |   TProperties extends object,
     |                       ^^^^^^
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:64:23 - error deprecated: Deprecated: object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.
> 64 |   TProperties extends object,
     |                       ^^^^^^
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:141:23 - error deprecated: Deprecated: object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.
> 141 |   TPatchModel extends object = TagsUpdateModel<TResource>,
      |                       ^^^^^^
/mnt/vss/_work/1/s/specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:91:23 - error deprecated: Deprecated: object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.
> 91 |   TProperties extends object,
     |                       ^^^^^^
```

**Comment by mikeharder** [comment]
> @mikeharder, @markcowl, it looks like the TSP Validation step is resolving the pipeline working path for `common-types` to [agent build path](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2840004&view=logs&j=33174e5f-e491-5dfc-305d-c09df824202b&t=0f377f82-3711-5097-078f-1ce87e974d2b&l=561).

Yes, this is known bug in the old TypeSpecValidation tool.  We are trying to replace the old tool ASAP with a re-write based on the logic from the new CI pipeline.

**Comment by mikeharder** [comment]
CI is still failing with many instances of these two errors:

```
specification/containerstorage/ContainerStorage.Management/custom_operations.tsp:106:1
error @azure-tools/typespec-azure-resource-manager/arm-resource-operation-outside-interface: 
All operations must be inside an interface declaration.

specification/containerstorage/ContainerStorage.Management/main.tsp:227:3
error @azure-tools/typespec-azure-core/documentation-required: The EnumMember named
'Succeeded' should have a documentation or description, please use decorator @doc to add it.
```

For the doc-required errors, we can either add the docs ourselves, or add the following to suppress for now and the service team should fix later:

```
#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=2840217&view=logs&j=9cd39ef8-47c1-5b58-08db-8c3708712f1b&t=fc9dfc5e-afbb-5dcd-1f9e-c63049dbf79d

**Comment by v-xuto** [comment]
@allenjzhang Please fix CI checks errors (`Model validation`), you can refer to this [comments](https://github.com/Azure/azure-rest-api-specs/pull/24397#issuecomment-1587871013) or  https://aka.ms/ci-fix on how to fix it.

**Comment by allenjzhang** [comment]
@mikeharder , @markcowl , @rnhan , the migration is finally ready for review.

**Comment by allenjzhang** [comment]
> @allenjzhang: CI is still failing because the generated swagger doesn't match the PR. I think you just need to re-generate swagger and update the PR, then this should be passing CI and ready to merge from my perspective.
> 
> After CI is passing, someone should review the failures in "Swagger Breaking Change" and "Swagger LintDiff".

Done. Breaking change is accepted.

**Review by mikeharder — APPROVED** [review]

**Pull request merged**